stage0/run.go

```diff
@@ -308,6 +308,15 @@ func Run(cfg RunConfig, dir string) {
 		log.Fatalf("failed changing to dir: %v", err)
 	}
 
+	// we write /etc/machine-id here because systemd-nspawn needs it to link
```

Comment:
I have to look more into this but AFAIU systemd does that for us but we need that before systemd starts: in systemd-nspawn. It reads it to compare it with host's /etc/machine-id and refuse to start if they match.